doc/filters/format_currency.rst

```diff
@@ -20,25 +20,75 @@ You can pass attributes to tweak the output:
 
-* ``grouping_used``;
-* ``decimal_always_shown``;
-* ``max_integer_digit``;
-* ``min_integer_digit``;
-* ``integer_digit``;
-* ``max_fraction_digit``;
-* ``min_fraction_digit``;
-* ``fraction_digit``;
-* ``multiplier``;
-* ``grouping_size``;
-* ``rounding_mode``;
-* ``rounding_increment``;
-* ``format_width``;
-* ``padding_position``;
-* ``secondary_grouping_size``;
-* ``significant_digits_used``;
-* ``min_significant_digits_used``;
-* ``max_significant_digits_used``;
-* ``lenient_parse``.
+* ``grouping_used``: Specifies whether to use grouping separator for thousands.
+    .. code-block:: twig
+
+        {# €1,234,567.89 #}
+        {{ 1234567.89 | format_currency('EUR', {grouping_used:true}, 'en') }}
+
+* ``decimal_always_shown``: Specifies whether to always show the decimal part, even if it's zero.
+    .. code-block:: twig
+
+        {# €123.00 #}
+        {{ 123 | format_currency('EUR', {decimal_always_shown:true}, 'en') }}
+
+* ``max_integer_digit``:
+* ``min_integer_digit``:
+* ``integer_digit``: Define constraints on the integer part.
+    .. code-block:: twig
+
+        {# €345.68 #}
+        {{ 12345.6789 | format_currency('EUR', {max_integer_digit:3, min_integer_digit:2}, 'en') }}
+
+* ``max_fraction_digit``:
+* ``min_fraction_digit``:
+* ``fraction_digit``: Define constraints on the fraction part.
+    .. code-block:: twig
+
+        {# €123.46 #}
+        {{ 123.456789 | format_currency('EUR', {max_fraction_digit:2, min_fraction_digit:1}, 'en') }}
+
+* ``multiplier``: Multiplies the value before formatting.
+    .. code-block:: twig
+
+        {# €123,000.00 #}
+        {{ 123 | format_currency('EUR', {multiplier:1000}, 'en') }}
+
+* ``grouping_size``:
+* ``secondary_grouping_size``: Set the size of the primary and secondary grouping separators.
+    .. code-block:: twig
+
+        {# €1,23,45,678.00 #}
+        {{ 12345678 | format_currency('EUR', {grouping_size:3, secondary_grouping_size:2}, 'en') }}
+
+* ``rounding_mode``:
+* ``rounding_increment``: Control rounding behavior.
+    .. code-block:: twig
+
+        {# €123.50 #}
+        {{ 123.456 | format_currency('EUR', {rounding_mode:'ceiling', rounding_increment:0.05}, 'en') }}
+
+* ``format_width``:
+* ``padding_position``: Set width and padding for the formatted number.
+    .. code-block:: twig
+
+        {# €123.00 #}
+        {{ 123 | format_currency('EUR', {format_width:10, padding_position:'before_suffix'}, 'en') }}
+
+* ``significant_digits_used``:
+* ``min_significant_digits_used``:
+* ``max_significant_digits_used``: Control significant digits in formatting.
+    .. code-block:: twig
+
+        {# €123.4568 #}
+        {{ 123.456789 | format_currency('EUR', {significant_digits_used:true, min_significant_digits_used:4, max_significant_digits_used:7}, 'en') }}
+
+
+* ``lenient_parse``: If true, allows lenient parsing of the input.
+    .. code-block:: twig
+
+        {# €123.00 #}
+        {{ 123 | format_currency('EUR', {lenient_parse:true}, 'en') }}
```

Comment:
Could be more descriptive of the lenient parsing effect

Reply:
I couldn't find a useful example, do you have any ideas ?

Reply:
I'm not sure here, to be honest. As i see it, the "lenient_parse" is a way to allow data that is not perfectly in the expected syntax... 


```php
<?php

$formatter = NumberFormatter::create('fr-FR', NumberFormatter::CURRENCY);

foreach ([false, true] as $lenientParse) {

    echo 'lenientParse: ' . ($lenientParse ? 'true' : 'false') . PHP_EOL;
    $formatter->setAttribute(NumberFormatter::LENIENT_PARSE, $lenientParse);

    echo $formatter->format(-9) . PHP_EOL;
    echo $formatter->format('.9') . PHP_EOL;
    echo $formatter->format(12.34000) . PHP_EOL;

    echo $formatter->format($formatter->parse('56,7800000')) . PHP_EOL;
    echo $formatter->format($formatter->parse('56,78')) . PHP_EOL;
    echo $formatter->format($formatter->parse('56,78€')) . PHP_EOL;
    echo $formatter->format($formatter->parse('56,78 €')) . PHP_EOL;
    echo $formatter->format($formatter->parse('56,78 €')) . PHP_EOL;
}
```

Would render 

```diff
lenientParse: false
 -9,00 €
0,90 €
12,34 €
-0,00 €
-0,00 €
-0,00 €
-0,00 €
56,78 €

lenientParse: true
 -9,00 €
0,90 €
12,34 €
+56,78 €
+56,78 €
+56,78 €
+56,78 €
56,78 €
```


As you see, it's when the `parse` method is called it makes a difference.

But it this specific case (currency), as we do not call "`parse`", and the `format` method only accept `int|float`, i don't see a concrete exemple where it would have an impact.